### DISCUSSION TITLE
Singleton sub-orchestrations

### DISCUSSION THREAD

**Discussion by obesser** [Azure/azure-functions-durable-extension#2474] [Q&A] [answered]
Hi,

Is there a way to achieve a singleton behavior for sub-orchestrations?

**Comment by ltouro** [comment]
Yes. Use a constant workflow ID. Only one will be ever executing with that ID.

**Reply by obesser** [reply]
But is there a way to check the status of a sub-orchestration, like in the example here https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-singletons?tabs=python#singleton-example

**Reply by ltouro** [reply]
Yes, a sub-orchestration is still a orchestration and you can use regular APIs to query it. 

Two notes:

I think there is a feature that auto-cancels the sub orchestration when the main one is cancelled (maybe even if they complete?). I never checked that.

When you let the SDK choose the instance ID, it will use the main orchestration ID + the activity index. I noted that such ID schema tends to make the query slower (at least using the VS Code Extension and with a huge history table). Maybe that have something to do with above feature and using a default ID may break it. Never checked that also.

**Reply by obesser** [reply]
Thanks @ltouro 

> Yes, a sub-orchestration is still a orchestration and you can use regular APIs to query it.

The status query in the example I referenced is from a client function (`client.get_status()`), how do I call this API from within the main orchestration?

**Reply by ltouro** [reply, marked as answer]
If you want to use this API, this should definitely be in an activity. You can plug the Client in the DI and pull it up in any activity class.

For tracking inside the orchestration, you should use the task that you are given when you start it, e.g. for blocking/unblocking the orchestration execution
